src/graphMethods/computeEpicenter.py

```diff
@@ -0,0 +1,132 @@
+with open('/home/scampit/software/MetOncoFit/srv/media.txt') as f:
+    media = f.readlines()
+media = [x.strip() for x in media]
+
+biomass_rxn = list()
+medium_rxn = list()
+biomass_dist = list()
+medium_dist = list()
+biomass_component = list()
+medium_component = list()
+# Get the shortest distance from the medium or biomass component to any given reaction.
+print("Computing medium and biomass epicenter scores")
+for node, _, metadata in recon1Graph.edges(data=True):
+    for b in biomass:
+        if (node.strip() or b.strip()) not in cofactors:
+            try:
+                # Distance from any given reaction to a biomass component
+                biomass_dist.append(len(nx.shortest_path(G=recon1Graph, source=node.strip(), target=b.strip()))  )
+                biomass_rxn.append(metadata['label'])
+                biomass_component.append(b.strip())
+
+            except: # If doesn't exist, set to 0
+                biomass_rxn.append(metadata['label'])
+                biomass_dist.append(0)
+                biomass_component.append(b.strip())
+
+    for m in media:
+        if (node.strip() or m.strip()) not in cofactors:
+            try:
+                # Distance from any given medium component to a reaction
+                medium_dist.append(len(nx.shortest_path(G=recon1Graph, source=m.strip(), target=node.strip())))
+                medium_rxn.append(metadata['label'])
+                medium_component.append(m.strip())
+
+            except: # If doesn't exist, set to 0
+                medium_rxn.append(metadata['label'])
+                medium_dist.append(0)
+                medium_component.append(m.strip())
+
+# Construct dataframes for the distance metrics
+print("Mapping scores from reactions to genes")
+biomass_df = pd.DataFrame({"Distance": biomass_dist,
+                           "Reaction": biomass_rxn,
+                           'Biomass': biomass_component})
+medium_df = pd.DataFrame({"Distance": medium_dist,
+                          "Reaction": medium_rxn,
+                          'Medium': medium_component})
+
+# Get the minimum shortest distance that is not 0 based on reaction-medium pairs --> bottleneck
+medium_df = medium_df.groupby(['Reaction', 'Medium']).apply(lambda x: x[x > 0].min(axis=1))
+medium_df = medium_df.reset_index().fillna(0)
+
+biomass_df = biomass_df.groupby(['Reaction', 'Biomass']).apply(lambda x: x[x > 0].min(axis=1))
+biomass_df = biomass_df.reset_index().fillna(0)
+
+# Map back to genes
+geneRxnMap = pd.read_csv('/home/scampit/software/MetOncoFit/srv/RECON1ReactionGeneMap.txt')
+medium_df = pd.merge(medium_df, geneRxnMap,
+                     how='inner',
+                     left_on='Reaction', right_on='Reaction')
+biomass_df = pd.merge(biomass_df, geneRxnMap,
+                     how='inner',
+                     left_on='Reaction', right_on='Reaction')
+print(biomass_df)
+
+# Pivot so that genes are rows and medium components are columns
```

Comment:
Section doesn't work due to duplicate genes. Need to figure out a way to aggregate based on unique gene-reaction-metabolite relationship.